### PR TITLE
update dependency

### PR DESCRIPTION
**Pull request by syaiful6**
- update free to 3.1.0
- update transformer 2.0.2

### CODE REVIEW

**Comment by paf31** [comment]
With the latest `free`, we can use the `explore` function to implement the free-cofree pairing actually.

**Comment by syaiful6** [comment]
hmm, i think ```explore``` defined on ```cofree``` have specialized version rather than pairing defined here, it symiliar to ```zap``` i think. i might be wrong, i just play around with cofree as interpreter. :+1:

**Comment by paf31** [comment]
Yes, but the pairing can be defined using `explore`, and it's safer.

**Comment by syaiful6** [comment]
@paf31 ok i implements it using ```explore```, could you review it?

File: src/Data/Functor/Pairing.purs [hunk 1]
```diff
@@ -76,8 +76,5 @@ writerTraced pairing f (WriterT writer) (TracedT gf) =
   pairing (\(Tuple a w) f1 -> f a (f1 w)) writer gf
 
 -- | `Free` pairs with `Cofree`.
-freeCofree :: forall f g. Functor f => f ⋈ g -> Free f ⋈ Cofree g
-freeCofree pairing f free cofree =
-  case resume free of
-    Left fa -> pairing (freeCofree pairing f) fa (tail cofree)
-    Right a -> f a (head cofree)
+freeCofree :: forall f g. (Functor f, Functor g) => f ⋈ g -> Free f ⋈ Cofree g
+freeCofree pairing f free cofree = explore (zap pairing) (map f free) cofree
```
Comment (paf31) [inline src/Data/Functor/Pairing.purs]: Nitpick: you can eta reduce this.

**Review by paf31 — CHANGES_REQUESTED** [review]
Looks good otherwise, thanks!

File: src/Data/Functor/Pairing.purs (hunk 1 above)
Comment (syaiful6) [reply src/Data/Functor/Pairing.purs]: thanks!

**Comment by syaiful6** [comment]
@paf31 thanks, i am use compose now. and my commit was squashed

**Comment by paf31** [comment]
Looks good, but can you please see if it's possible to do this without adding the new constraint?

**Comment by syaiful6** [comment]
i add it because ```explore``` require both ```f``` and ```g``` are Functor, otherwise it will fail to compile telling me the compiler cannot find an functor instance.

**Review by paf31 — APPROVED** [review]

**Comment by paf31** [comment]
Sorry, you're right, thanks!

**Pull request merged**